feat(#2318): install CLI by default

lsanpablo opened the pull request:
### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

- Install required CLI dependencies and CLI script as part of the base package
- Update docs to reflect above
- Add `jsbeautifier` group
- Handle missing optional dependencies gracefully in the `litestar` cli
- Add deprecation message to the CLI group in the docs

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

- Closes #2318

provinzkraut commented:
There are a few things:

1. Do we want to include `uvicorn` by default? I think we should not, and instead revisit support for different servers in the `run` command
2. Should `jsbeautifier` be included by default? IMO we can have an extra for this, and if it's not present, just don't use it. It can go in the `standard` extra.
3. The `cli` group should remain for now, to not break things. Not sure if poetry allows empty groups though

@litestar-org/maintainers @litestar-org/members thoughts?

peterschutt commented:
> Should jsbeautifier be included by default? IMO we can have an extra for this, and if it's not present, just don't use it. It can go in the standard extra.

+1 for not default

JacobCoffee commented:
>     1. Do we want to include `uvicorn` by default? I think we should not, and instead revisit support for different servers in the `run` command
No

>     2. Should `jsbeautifier` be included by default? IMO we can have an extra for this, and if it's not present, just don't use it. It can go in the `standard` extra.
No

>     3. The `cli` group should remain for now, to not break things. Not sure if poetry allows empty groups though
Agreed until V3

JacobCoffee commented:
@lsanpablo Please see the Sourcery suggested refactorings:

https://github.com/litestar-org/litestar/pull/2356/commits/cdff023f688af866464f91371019f1991ba678cd

lsanpablo commented:
Thanks for the feedback. I made the following changes:
- re-added the `cli` group which includes `jsbeautifier` and `uvicorn` to offer equivalent functionality to what it was before
- refactored the `litestar run` and `litestar schema typescript` commands to gracefully handle the fact that the script may be installed but `uvicorn` and `jsbeautifier` might not
- updated the documentation accordingly

JacobCoffee commented on docs/index.rst:
```diff
@@ -57,6 +54,10 @@ Installation
         :code:`pip install litestar[sqlalchemy]`
 
     :doc:`CLI </usage/cli>`
+        DEPRECATED (to be removed in v3) - Base installation now includes the CLI dependencies
+        and this group is no longer required to use the CLI. If you need the optional CLI dependencies,
+        install the ``standard`` group instead.
+
```
Can we use the `.. deprecated::` [directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) here? I think it would look better.
```suggestion
        
```
```suggestion
        .. deprecated:: 3.0.0
        
        The `litestar` base installation now includes the CLI dependencies and this group is no longer required 
        to use the CLI. 
        If you need the optional CLI dependencies,install the ``standard`` group instead.

```

JacobCoffee commented on docs/index.rst:
```diff
@@ -65,7 +66,7 @@ Installation
         :code:`pip install litestar[mako]`
 
-    Standard Installation (includes CLI and Jinja templating):
+    Standard Installation (includes uvicorn and Jinja templating):
```
```suggestion
    Standard Installation (includes Uvicorn and Jinja2 templating):
```

JacobCoffee commented on litestar/cli/commands/core.py:
```diff
@@ -95,7 +97,7 @@ def run_command(
     ctx: Context,
 ) -> None:
-    """Run a Litestar app.
+    """Run a Litestar app; requires uvicorn.
```
```suggestion
    """Run a Litestar app; requires ``uvicorn``.
```

JacobCoffee commented:
Looks good except for a few comments

JacobCoffee commented on docs/index.rst:
```diff
@@ -57,6 +54,12 @@ Installation
         :code:`pip install litestar[sqlalchemy]`
 
     :doc:`CLI </usage/cli>`
+        .. deprecated:: 2.1.1
```
Do we add a "will be removed in 3.0"?

provinzkraut replied:
We should, yea

lsanpablo commented:
Setting PR to draft while I resolve the `poetry.lock` merge conflict.

JacobCoffee commented:
lgtm. sorry for all the nits 😛

JacobCoffee commented:
@all-contributors add @lsanpablo code tests docs
